src/clj/game/core/moving.clj

```diff
@@ -457,12 +457,41 @@
   (let [cards (take n (shuffle (get-in @state [to-side :hand])))]
     (trash-cards state from-side eid cards {:unpreventable true})))
 
+(defn swap-legal?
+  "checks that swapping two installed corp cards is legal"
+  [state side a b]
+  (let [pred? (every-pred corp? installed?)
+        xor? (fn [a b f]
+               (or (and (f a) (not (f b)))
+                   (and (f b) (not (f a)))))]
```

Comment:
Maybe move the two `f` calls into a let: `(let [ret-a (f a) ret-b (f b)] (or (and ret-a (not ret-b)) (and (not ret-a) ret-b)))`? i'm not sure that's right, but it's close.

Reply:
Perhaps a less ugly approach would be something like:
`(= 1 (count (filter identity (apply f [a b]))))`

Reply:
That's significantly less readable, please no.